examples/step-63/step-63.cc

```diff
@@ -19,16 +19,1091 @@
+    if (dim > 1)
+      advection_direction[1] = std::cos(numbers::PI / 6.0);
+    if (dim > 2)
+      advection_direction[2] = std::sin(numbers::PI / 6.0);
+  }
+
+
+
+  template <int dim>
+  void AdvectionProblem<dim>::setup_system()
+  {
+    const unsigned int n_levels = triangulation.n_levels();
+
+    dof_handler.distribute_dofs(fe);
+
+    // We could renumber the active dofs with DoFRenumbering::downstream()
+    // here, but the smoothers only act on multigrid levels and as such, this
+    // wouldn't matter. Instead, we will renumber the DoFs on each multigrid
+    // level below.
+
+    solution.reinit(dof_handler.n_dofs());
+    system_rhs.reinit(dof_handler.n_dofs());
+
+    constraints.clear();
+    DoFTools::make_hanging_node_constraints(dof_handler, constraints);
+
+    VectorTools::interpolate_boundary_values(
+      mapping, dof_handler, 0, BoundaryValues<dim>(), constraints);
+    VectorTools::interpolate_boundary_values(
+      mapping, dof_handler, 1, BoundaryValues<dim>(), constraints);
+    constraints.close();
+
+    DynamicSparsityPattern dsp(dof_handler.n_dofs());
+    DoFTools::make_sparsity_pattern(dof_handler,
+                                    dsp,
+                                    constraints,
+                                    /*keep_constrained_dofs = */ false);
+
+    sparsity_pattern.copy_from(dsp);
+
+    system_matrix.reinit(sparsity_pattern);
+
+
+    // Setup GMG DoFs
+    dof_handler.distribute_mg_dofs();
+
+    // Renumber DoFs on each level in downstream or upstream direction if
+    // needed. This is only necessary for point smoothers (SOR and Jacobi) as
+    // the block smoothers operate on cells (see create_smoother()):
+    if (settings.smoother_type == "sor" || settings.smoother_type == "jacobi")
+      {
+        if (settings.dof_renum == "downstream" ||
+            settings.dof_renum == "upstream")
+          {
+            const Tensor<1, dim> direction =
+              (settings.dof_renum == "upstream" ? -1.0 : 1.0) *
+              advection_direction;
+
+            for (unsigned int level = 0; level < n_levels; ++level)
+              DoFRenumbering::downstream(dof_handler,
+                                         level,
+                                         direction,
+                                         /*dof_wise_renumbering = */ true);
+          }
+        else if (settings.dof_renum == "random")
+          {
+            for (unsigned int level = 0; level < n_levels; ++level)
+              DoFRenumbering::random(dof_handler, level);
+          }
+        else
+          Assert(false, ExcNotImplemented());
+      }
+
+    mg_constrained_dofs.clear();
+    mg_constrained_dofs.initialize(dof_handler);
+
+    std::set<types::boundary_id> dirichlet_boundary_ids = {0, 1};
+    mg_constrained_dofs.make_zero_boundary_constraints(dof_handler,
+                                                       dirichlet_boundary_ids);
```

Comment:
you can avoid the local variable by just putting `{0, 1}` in line 603 here. If you want to give the variable a name, mark it `const`.